Fixing Pinterest Save button positioning for Twenty Twenty-Four theme.

message-dimke opened the pull request:
### Changes proposed in this Pull Request:

Closes #1054.

Pinterest Save button has CSS absolute positioning when the product card has no CSS relative position rule on it which makes Pinterest Save button align relatively to the document, instead of a product card element.

### Screenshots:

<img width="1811" alt="Cursor_and_Shop_–_Wildly_Involved" src="https://github.com/user-attachments/assets/ff6c1d2a-8edf-4e39-bbc7-eefdf84e3dd5">

<img width="1530" alt="Music_–_Wildly_Involved" src="https://github.com/user-attachments/assets/7b357beb-f77b-45cf-bd28-8266b731d11c">

### Detailed test instructions:

1. Install Pinterest extension with Twenty Twenty-Four.
2. Navigate to the Shop.
3. Hover any product card.
4. Observe Pinterest Save button is out of bounds.

### Changelog entry

> Fix - Pinterest Save button positioning.

message-dimke committed:
Adding relative product image positioning so the Pinterest save button with its absolute positioning would stay above the product card.

ecgan commented on assets/source/sass/frontend/pinterest-for-woocommerce-pins.scss:
```diff
@@ -24,10 +24,12 @@
 
 // Twenty Twenty-Four compatibility.
-.wp-block-post {
+.wp-block-product {
```
I believe you mean this?

```suggestion
.wc-block-product {
```

ecgan replied:
@message-dimke , do you happen to know when and where `wp-block-post` is changed to `wc-block-product`? I was searching around but couldn't find the info.

Also, maybe for backward compatibility, we need to have both, like the following, in case some users are using some older versions? (need to test this out.)

```suggestion
.wp-block-post,
.wc-block-product {
```

ecgan reviewed (changes requested):
@message-dimke , leaving one finding below.

ecgan commented:
@message-dimke , I was testing around and I noticed that the Pinterest "Save" button does not show up in the product page with Twenty Twenty-Four theme.

Screenshot of product image in product page:

![image](https://github.com/user-attachments/assets/bb9d000b-986c-4fe1-b792-cb35dbccf3fc)

Screenshot of related products in product page - note that for this one, it is using `wp-block-post` CSS class, not `wc-block-product`, and it does not have the `pinterest-for-woocommerce-image-wrapper` div:

![image](https://github.com/user-attachments/assets/5fdc2a20-1373-478e-86b7-5ce41de0d9c3)

Maybe you could consider opening separate issues / PRs to fix them.

message-dimke committed:
Update assets/source/sass/frontend/pinterest-for-woocommerce-pins.scss

Fixing class name typo and adding wp-block-post back to support related products section.

Co-authored-by: Gan Eng Chin <ecgan@users.noreply.github.com>

message-dimke committed:
Updating css rules to render Pinterest Save button properly.
